angular/openmrs-search/openmrs-search.component.js

```diff
@@ -0,0 +1,225 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+var template = require('./openmrs-search.html');
+var openmrs = require('../../src/scss/images/inprogress.gif'); //TODO: *.png logo image here
+import openmrsRest from './../openmrs-rest/openmrs-rest.js';
+
+export default angular.module('openmrs-contrib-uicommons.search', ['openmrs-contrib-uicommons.rest'])
+    .component('openmrsSearch', {
+        template: template,
+        controller: openmrsSearch,
+        controllerAs: 'vm',
+        bindings: {
+            resource: '<',
+            columnsMain: '<',
+            columnsDetails: '<',
+            redirectionParam: '<',
```

Comment:
That is fine. It is called 'default' so that it can be overwritten when necessary :) The point is to make it possible to use less config if you stick to conventions.